### PR TITLE
Ccw4057

### PR DESCRIPTION
**Pull request by nke001**
Solving flake 8 errors for theano/tensor/nnet/test_*

fix gh-4057

### CODE REVIEW

**Comment by anirudh9119** [comment]
@adri-romsor
@nouiz

File: theano/tensor/nnet/tests/test_nnet.py [hunk 1]
```diff
@@ -1598,15 +1604,6 @@ def test_relu():
         y = relu(x, alpha).eval({x: X, alpha: A})
         assert numpy.allclose(y, numpy.where(X > 0, X, A * X), rtol=3e-5)
 
-    # test that for alpha of ndarray don't cause upcast.
```
Comment (nouiz) [inline theano/tensor/nnet/tests/test_nnet.py]: why remove this ? I think that removing this remove test for corner case.

File: theano/tensor/nnet/tests/test_sigm.py [hunk 2]
```diff
@@ -339,7 +339,7 @@ def test_local_ultra_fast_sigmoid(self):
         topo = f.maker.fgraph.toposort()
         assert topo[0].op == ultra_fast_sigmoid
         assert len(topo) == 1
-        ux_v = f([[-50, -10, -4, -1, 0, 1, 4, 10, 50]])
+        # ux_v = f([[-50, -10, -4, -1, 0, 1, 4, 10, 50]])
```
Comment (nouiz) [inline theano/tensor/nnet/tests/test_sigm.py]: uncomment that. We want the call to be done. Just don't affect it to a variable.

File: theano/tensor/nnet/tests/test_sigm.py [hunk 3]
```diff
@@ -358,7 +358,7 @@ def test_local_hard_sigmoid(self):
         topo = f.maker.fgraph.toposort()
         assert len(topo) > 1
         assert not any([n.op == sigmoid for n in topo])
-        ux_v = f([[-50, -10, -4, -1, 0, 1, 4, 10, 50]])
+        # ux_v = f([[-50, -10, -4, -1, 0, 1, 4, 10, 50]])
```
Comment (nouiz) [inline theano/tensor/nnet/tests/test_sigm.py]: idem

File: theano/tensor/nnet/tests/test_sigm.py [hunk 4]
```diff
@@ -292,10 +292,10 @@ def ok(expr1, expr2):
         ok(-sigmoid(-x) *
            (exp(y) * (-exp(-z) * 3 * -exp(x)) *
             (y * 2 * (-sigmoid(-y) * (z + t) * exp(z)) * sigmoid(z))) *
-           -sigmoid(x),
+           (-sigmoid(x)),
```
Comment (nouiz) [inline theano/tensor/nnet/tests/test_sigm.py]: I'm curious, why this change?

File: theano/tensor/nnet/tests/test_sigm.py (hunk 2 above)
Comment (nke001) [inline theano/tensor/nnet/tests/test_sigm.py]: changed it to 
f([[-50, -10, -4, -1, 0, 1, 4, 10, 50]])

File: theano/tensor/nnet/tests/test_sigm.py (hunk 3 above)
Comment (nke001) [inline theano/tensor/nnet/tests/test_sigm.py]: changed to f([[-50, -10, -4, -1, 0, 1, 4, 10, 50]])

File: theano/tensor/nnet/tests/test_sigm.py (hunk 4 above)
Comment (nke001) [inline theano/tensor/nnet/tests/test_sigm.py]: changed it back

**Comment by nouiz** [comment]
Why did you rebase? It broke the history again...

File: theano/tensor/nnet/tests/test_sigm.py [hunk 5]
```diff
@@ -286,8 +286,8 @@ def ok(expr1, expr2):
            -x * sigmoid(-x) * (y * (-1 * z)))
         ok(-sigmoid(-x) *
            (exp(y) * (-exp(-z) * 3 * -exp(x)) *
-            (y * 2 * (-sigmoid(-y) * (z + t) * exp(z)) * sigmoid(z))) *
-           (-sigmoid(x)),
+            (y * 2 * (-sigmoid(-y) * (z + t) * exp(z)) * sigmoid(z))) * -
+           sigmoid(x),
```
Comment (nouiz) [inline theano/tensor/nnet/tests/test_sigm.py]: Having the - at the end make this less readable.

**Comment by nke001** [comment]
I rebased, was it not done correctly? I will probably ask when I'm in the
lab then

**Comment by abergeron** [comment]
You should probably also remove the files you fixed from the whitelist here: https://github.com/Theano/Theano/blob/master/theano/tests/test_flake8.py#L76

**Comment by nouiz** [comment]
Now there is this failing. Just add the missing import:

/home/travis/build/Theano/Theano/theano/tensor/nnet/tests/__init__.py:1:1: FI10 __future__ import "division" missing

/home/travis/build/Theano/Theano/theano/tensor/nnet/tests/__init__.py:1:1: FI11 __future__ import "absolute_import" missing

/home/travis/build/Theano/Theano/theano/tensor/nnet/tests/__init__.py:1:1: FI13 __future__ import "print_function" missing

To run locally our test_flake8.py (and not the command line flake8), you can do it like this:

nosetests theano/tests/test_flake8.py

File: theano/tests/test_flake8.py [hunk 6]
```diff
@@ -72,14 +72,6 @@
     "tensor/tests/test_blas_scipy.py",
     "tensor/tests/test_mpi.py",
     "tensor/nnet/__init__.py",
-    "tensor/nnet/tests/__init__.py",
```
Comment (abergeron) [inline theano/tests/test_flake8.py]: Leave the `__init__` file in the whitelist.  It's too hard to clean up properly.

Comment (nouiz) [inline theano/tests/test_flake8.py]: I agree with @abergeron.
